DashAISoftware/DashAI
# Project/datasets rework

# Summary


This PR contains all the changes of the Datasets Rework project that disconnect the datasets loading to the task that the user wants to solve, now the task is setted on the experiment configuration.

## Type of change



- Front end new feature.
- Back end new feature.
- Refactoring.

## Changes


Changes on dataset loading flow:
- Refactoring: Deletes the step where the user select the Task on the dataset uploading.
- New feature: Adds a new step with a summary of the dataset that contains columns and their types.
![Endpoints API (4)](https://github.com/DashAISoftware/DashAI/assets/70553082/fb580034-52e6-498f-982d-0049f4a5a9b4)

Changes on experiment creation flow:
- Refactoring: Now the step of dataset selection shows all the datasets.
- New feature: Adds a new step with the dataset preparation where the user can choose wich columns to use and how to split the dataset.
![Endpoints API (3)](https://github.com/DashAISoftware/DashAI/assets/70553082/1f354187-8e47-465d-8d74-386a1af133d8)

This PR also has all the refactoring needed to delete the input and output columns definition and usage on the DashAI Datasets.

## How to Test


To check the new features:
1.- Run ```python -m DashAI```
2.- Go to ```dashai/front``` and run ```yarn start```
3.- Go to the datasets page and upload a new dataset.
4.- Then go to the experiements page and create a new one.

## Screenshots



Datasets new step: Dataset Summary
![image](https://github.com/DashAISoftware/DashAI/assets/70553082/85e42fcd-908a-4647-91e2-93ffa2467bd2)

Experiments new step: Prepare Dataset
![image](https://github.com/DashAISoftware/DashAI/assets/70553082/a43b7d63-fe53-48ee-ad09-a5fbacece19d)
